go/ct/gen/state_test.go

```diff
@@ -478,3 +506,149 @@ func TestStateGenerator_BlockNumberHashes(t *testing.T) {
+// Storage
+
+func TestStateGenerator_BindStorageConfiguration(t *testing.T) {
+	generator := NewStateGenerator()
+	generator.BindStorageConfiguration(tosca.StorageAdded, Variable("a"), Variable("b"))
+	constraint := storageConfigConstraint{tosca.StorageAdded, Variable("a"), Variable("b")}
+	if !slices.Contains(generator.storageGen.cfg, constraint) {
+		t.Errorf("unexpected storage configuration")
+	}
+}
+
+func TestStateGenerator_BindIsStorageCold(t *testing.T) {
+	generator := NewStateGenerator()
+	generator.BindIsStorageCold(Variable("a"))
+	constraint := warmColdConstraint{Variable("a"), false}
+	if !slices.Contains(generator.storageGen.warmCold, constraint) {
+		t.Errorf("unexpected storage configuration")
+	}
+}
+
+// //////////////////////////////////////////////////////////
+// Transient storage
+func TestStateGenerator_BindTransientStorageToNonZero(t *testing.T) {
+	generator := NewStateGenerator()
+	generator.BindTransientStorageToNonZero(Variable("a"))
+	if !generator.transientStorageGen.nonZeroConstraints[Variable("a")] {
+		t.Errorf("unexpected transient storage configuration")
+	}
+}
+
+// //////////////////////////////////////////////////////////
+// Accounts
+func TestStateGenerator_BindToColdAddress(t *testing.T) {
+	generator := NewStateGenerator()
+	generator.BindToColdAddress(Variable("a"))
+	constraint := warmColdConstraint{Variable("a"), false}
+	if !slices.Contains(generator.accountsGen.warmCold, constraint) {
+		t.Errorf("unexpected account configuration")
+	}
+}
+
+////////////////////////////////////////////////////////////
+// block hash list
+
+func TestStateGenerator_RestrictVariableToOneOfTheLast256Blocks(t *testing.T) {
+	generator := NewStateGenerator()
+	generator.RestrictVariableToOneOfTheLast256Blocks(Variable("a"))
+	if !generator.blockContextGen.rangeConstraints[Variable("a")] {
+		t.Errorf("unexpected block context configuration")
+	}
+}
+
+func TestStateGenerator_RestrictVariableToNoneOfTheLast256Blocks(t *testing.T) {
+	generator := NewStateGenerator()
+	generator.RestrictVariableToNoneOfTheLast256Blocks(Variable("a"))
+	if generator.blockContextGen.rangeConstraints[Variable("a")] {
+		t.Errorf("unexpected block context configuration")
+	}
+}
+
+////////////////////////////////////////////////////////////
+// block number offset
+
+func TestStateGenerator_SetBlockNumberOffsetValue(t *testing.T) {
+	generator := NewStateGenerator()
+	generator.SetBlockNumberOffsetValue(Variable("a"), 10)
+	if generator.blockContextGen.valueConstraint[Variable("a")] != 10 {
+		t.Errorf("unexpected block context configuration")
+	}
+}
+
+// //////////////////////////////////////////////////////////
+// blob hash index
+func TestStateGenerator_IsAbsentBlobHashIndex(t *testing.T) {
+	generator := NewStateGenerator()
+	generator.IsAbsentBlobHashIndex(Variable("a"))
+	if generator.transactionContextGen.blobHashVariables[Variable("a")] {
+		t.Errorf("unexpected transaction context configuration")
+	}
+}
```

Comment:
After reading all these generators, I have two suggestions:
- It may be possible to factor all of them in one or two groups, with table tests. 
- I am not very familiar with the generation process itself, but I think it may be possible to add constraints and generate a state. Where the constraints can be observed. I am a little surprised that that kind of test does not exist already, and if it does, it could be easy to upgrade it to cover these missing parts. 
   -  something like [this](https://github.com/Fantom-foundation/Tosca/blob/2972bf5e4475318ed368c39493a2e80befbe0d1f/go/ct/gen/code_test.go#L118)  or  [this](https://github.com/Fantom-foundation/Tosca/blob/2972bf5e4475318ed368c39493a2e80befbe0d1f/go/ct/gen/transaction_context_test.go#L98)

Reply:
done